ts/util/text-input.ts

```diff
@@ -15,7 +25,9 @@ export class TextInput {
     this.container.className =
       "font-inter font-medium text-12 text-black px-2 py-1 rounded-md ring-2 ring-offset-1 ring-red inline-block flex-auto items-center space-x-2";
     this.textField.className =
-      "ring-0 border-0 outline-0 rounded-sm relative z-10 px-1 w-4";
+      "ring-0 border-0 outline-0 rounded-sm relative z-10 px-1";
+    this.textField.style.minWidth = "20ch";
+    this.textField.style.width = "20ch";
```

Comment:
Shouldn't be using inline styles. Use Tailwind class to set minWidth and don't set width

Reply:
the snipped used to increase width dynamically doesn't play well with tailwind's min-w unfortunately :( Using .style is messier but it's the only way I've managed to get it to work

Reply:
I played around with it and found a solution that works without inline styles:

Here (in the constructor):

* add class `min-w-[150px]`
* set size: `this.textField.setAttribute("size", "10");`

In `keydown`:

* `this.textField.setAttribute("size", Math.max(this.textField.value.length, 10).toString());`